test/extended/prometheus/collection_profiles.go

```diff
@@ -0,0 +1,323 @@
+		}
+		r.pclient = oc.NewPrometheusClient(tctx)
+
+		var operatorConfiguration *v1.ConfigMap
+		o.Eventually(func() error {
+			operatorConfiguration, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+			return err
+		}).Should(o.BeNil())
+		r.originalOperatorConfiguration = operatorConfiguration
+	})
+
+	g.AfterAll(func() {
+		currentConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		o.Expect(err).To(o.BeNil())
+		currentConfiguration.Data = r.originalOperatorConfiguration.Data
+		_, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Update(tctx, currentConfiguration, metav1.UpdateOptions{})
+		o.Expect(err).To(o.BeNil())
+	})
+
+	var _ = g.Context("in an environment that migrates from default to minimal collection profile", func() {
+		g.It("should apply default collection profile initially", func() {
+			o.Eventually(func() error {
+				monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, collectionProfileDefault})
+				if err != nil {
+					return err
+				}
+				if len(monitors.Items) == 0 {
+					return fmt.Errorf("no monitors found with collection profile %q", collectionProfileDefault)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		})
+		g.It("should expose information about the applied collection profile using meta-metrics", func() {
+			vectorExpression := "profile:cluster_monitoring_operator_collection_profile:max{profile=\"%s\"} == 1"
+
+			for _, profile := range collectionProfilesSupportedList {
+				err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+				o.Expect(err).To(o.BeNil())
+
+				o.Eventually(func() error {
+					queryResponse, err := helper.RunQuery(tctx, r.pclient, fmt.Sprintf(vectorExpression, profile))
+					if err != nil {
+						return err
+					}
+					if len(queryResponse.Data.Result) == 0 {
+						return fmt.Errorf("no result found for profile %q", profile)
+					}
+
+					return nil
+				}).Should(o.BeNil())
+			}
+		})
+		g.It("should have at least one implementation for each collection profile", func() {
+			for _, profile := range collectionProfilesSupportedList {
+				err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+				o.Expect(err).To(o.BeNil())
+
+				o.Eventually(func() error {
+					monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, profile})
+					if err != nil {
+						return err
+					}
+					if len(monitors.Items) == 0 {
+						return fmt.Errorf("no monitors found with collection profile %q", profile)
+					}
+
+					return nil
+				}).Should(o.BeNil())
+			}
+		})
+		g.It("should hide all default-only metrics when minimal collection profile is enabled", func() {
+			appNameSelector := "app.kubernetes.io/name"
+			appName := "kube-state-metrics"
+			for _, profile := range []string{collectionProfileFull, collectionProfileMinimal} {
```

Comment:
Iiuc this is the test case to test applying the minimal profile (after full was applied). Why does this iterate over all profiles, including full?

Reply:
ACK, dropped the previous `full` profile safety checks since we've changed the approach.